examples/regression/3-scientific-articles/plot_zaffran2022_comparison.py

```diff
@@ -0,0 +1,282 @@
+
+train_size = all_x_train[0].shape[0]
+idx = np.array(range(train_size))
+n_half = int(np.floor(train_size/2))
+
+X_train_0 = X_train_0[:n_half]
+y_train_0 = y_train_0[:n_half]
+
+mapie_aci = mapie_aci.fit(X_train_0, y_train_0)
+y_pred_aci_npfit, y_pis_aci_npfit = mapie_aci.predict(
+    X_test_0, alpha=alpha, ensemble=True, optimize_beta=False
+)
+print("MAPIE estimator fitted!")
+
+# step 0
+y_pred_aci_pfit = np.zeros(y_pred_aci_npfit.shape)
+y_pis_aci_pfit = np.zeros(y_pis_aci_npfit.shape)
+y_pred_aci_pfit[:gap], y_pis_aci_pfit[:gap, :, :] = mapie_aci.predict(
+    X_test_0.iloc[:gap, :], alpha=alpha, ensemble=True, optimize_beta=False
+)
+
+# step t
+for step in range(1, min(len(X_test_0), iteration_max+1), gap):
+
+    mapie_aci.estimator_.single_estimator_.fit(
+        X_test_0.iloc[(step - gap):step, :],
+        y_test_0.iloc[(step - gap):step]
+    )
+
+    mapie_aci.partial_fit(
+        X_test.iloc[(step - gap):step, :],
+        y_test.iloc[(step - gap):step],
+    )
+
+    mapie_aci.adapt_conformal_inference(
+        X_test_0.iloc[(step - gap):step, :],
+        y_test_0.iloc[(step - gap):step],
+        gamma=gamma
+    )
+
+    (
+        y_pred_aci_pfit[step:step + gap],
+        y_pis_aci_pfit[step:step + gap, :, :],
+    ) = mapie_aci.predict(
+        X_test_0.iloc[step:(step + gap), :],
+        alpha=alpha,
+        ensemble=True,
+        optimize_beta=True
+    )
+
+results = y_pis_aci_pfit.copy()
+
+
+#########################################################
+# Get referenced result to reproduce
+#########################################################
+
+def get_pickle() -> Tuple[NDArray, NDArray]:
+    """
+    TODO
+
+    Returns
+    -------
+    TODO
+    """
+    website = "https://github.com/"
+    page = "mzaffran/AdaptiveConformalPredictionsTimeSeries/raw/"
+    folder = "131656fe4c25251bad745f52db3c2d7cb1c24bbb/results/"
+    folder += "Spot_France_Hour_0_train_2019-01-01/"
+    file = "ACP_0.04_RF.pkl"
+    url = website + page + folder + file
+    ssl._create_default_https_context = ssl._create_unverified_context
+    try:
+        loaded_data = pickle.load(urlopen(url))
+    except FileNotFoundError:
+        print(f"The file {file} was not found.")
+    except Exception as e:
+        print(f"An error occurred: {e}")
+    return loaded_data
+
+
+data_ref = get_pickle()
+
+
+#########################################################
+# Compare results
+#########################################################
+
+# Flatten the array to shape (n, 2)
+results_ref = np.concatenate(
+    [data_ref['Y_inf'], data_ref['Y_sup']], axis=0
+).T
+results = np.array(results.reshape(-1, 2))
+
+# Compare the NumPy array with the corresponding DataFrame columns
+comparison_result_Y_inf = np.array_equal(
+    results[:iteration_max, 0],
+    results_ref[:iteration_max, 0]
+)
+comparison_result_Y_sup = np.array_equal(
+    results[:iteration_max, 1],
+    results_ref[:iteration_max, 1]
+)
+
+# Print the comparison results
```

Comment:
We don't see any plots? This example only has printouts. I would at least reproduce graphs that are in the paper. Or show the results in a table manner.

Reply:
Done.